knack/config.py

```diff
@@ -47,36 +47,126 @@ def __init__(self, config_dir=None, config_env_var_prefix=None, config_file_name
         self._env_var_format = '{}{}'.format(env_var_prefix, '{section}_{option}')
         self.defaults_section_name = CLIConfig._CONFIG_DEFAULTS_SECTION
         self.config_parser.read(self.config_path)
+        self.config_file_chain = []
+        current_dir = os.getcwd()
+        config_dir_name = os.path.basename(self.config_dir)
+        while current_dir:
+            current_config_dir = os.path.join(current_dir, config_dir_name)
+            if (os.path.normcase(os.path.normpath(current_config_dir)) ==
+                    os.path.normcase(os.path.normpath(self.config_dir))):
+                break
+            if os.path.isdir(current_config_dir):
+                self.config_file_chain.append(_ConfigFile(current_config_dir,
+                                                          os.path.join(current_config_dir, configuration_file_name)))
+            if current_dir == os.path.dirname(current_dir):
+                break
+            current_dir = os.path.dirname(current_dir)
+        self.config_file_chain.append(_ConfigFile(self.config_dir, self.config_path))
 
     def env_var_name(self, section, option):
         return self._env_var_format.format(section=section.upper(),
                                            option=option.upper())
 
-    def has_option(self, section, option):
+    def has_option(self, section, option, use_local_config=False):
```

Comment:
It seems to me that whether the user is using local config should be a property of the CLIConfig object and set during `__init__` rather than specified on each and every method call. Is there some scenario where it would make sense to `get` a property from local config but set it NOT using the local context?  If that was the case, I would still suggest making `use_local_config=None` and then pick the default value from the CLIConfig's property.

Reply:
Updated.